### PR TITLE
Keep proxy in environment when using systemd-run

### PR DESCRIPTION
**Pull request by adlerweb**
Ref #101 
When running as root pikaur uses _systemd-run_ to isolate _makepkg_. _Systemd-run_ creates an empty environment and doesn't adhere to global profile settings like _/etc/profile.d/*_. These lines will add the variables neccesary for proxy usage to the temporary environment if they're set in the parent one.

### CODE REVIEW

**Comment by actionless** [comment]
could you please rebase on master?

arch linux base image path on dockerhub changed so travis is failing

File: pikaur/core.py [hunk 1]
```diff
@@ -144,12 +144,12 @@ def isolate_root_cmd(cmd: List[str], cwd=None) -> List[str]:
     ]
     if cwd is not None:
         base_root_isolator += ['-p', 'WorkingDirectory=' + cwd]
-    if "http_proxy" in os.environ:
-        base_root_isolator += ['-E', 'http_proxy=' + os.environ.get('http_proxy')]
-    if "https_proxy" in os.environ:
-        base_root_isolator += ['-E', 'https_proxy=' + os.environ.get('https_proxy')]
-    if "ftp_proxy" in os.environ:
-        base_root_isolator += ['-E', 'ftp_proxy=' + os.environ.get('ftp_proxy')]
+    if "http_proxy" in os.environ and os.environ.get('http_proxy') is not None:
```
Comment (actionless) [inline pikaur/core.py]: just `if os.environ.get('http_proxy') is not None:` will be enough

File: pikaur/core.py [hunk 2]
```diff
@@ -144,12 +144,12 @@ def isolate_root_cmd(cmd: List[str], cwd=None) -> List[str]:
...
+        base_root_isolator += ['-E', 'http_proxy=' + str(os.environ.get('http_proxy'))]
```
Comment (actionless) [inline pikaur/core.py]: just `os.environ['http_proxy']` should be fine

Comment (adlerweb) [reply pikaur/core.py]: I agree, it should according to the docs, however they seem to be not quite right, the checks don't work as expected . Without using str() MyPy failed over here when the variables where not set:

```
pikaur/core.py:148: error: Unsupported operand types for + ("str" and "None")
pikaur/core.py:148: note: Right operand is of type "Optional[str]"
```

Comment (actionless) [reply pikaur/core.py]: i don't think what the warning which you showing is generated for `os.environ['http_proxy']` line

File: pikaur/core.py (hunk 1 above)
Comment (actionless) [reply pikaur/core.py]: ^ this comment is still actual: https://travis-ci.com/actionless/pikaur/builds/108801760#L1424

**Pull request merged**